hack/gatewayapi-conformance.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Get Gateway API CRD bundle-version
+BUNDLE_VERSION=$(oc get crds gateways.gateway.networking.k8s.io -ojson | jq -r '.metadata.annotations."gateway.networking.k8s.io/bundle-version"')
+
+if [[ "${BUNDLE_VERSION}" == "null" ]]; then
+    echo "Cannot find bundle-version annotation from Gateway API CRD"
+    exit 1
+fi
+echo "Gateway API CRD bundle-version: ${BUNDLE_VERSION}"
+
+echo "Create GatewayClass gateway-conformance"
+oc apply -f -<<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: GatewayClass
+metadata:
+  name: gateway-conformance
+spec:
+  controllerName: openshift.io/gateway-controller
+EOF
+
+oc wait --for=condition=Accepted=true gatewayclass/gateway-conformance --timeout=300s
+
+echo "All gatewayclass status:"
+oc get gatewayclass -A
+
+echo "Go version: $(go version)"
+cd ../..
+mkdir -p kubernetes-sigs
+cd kubernetes-sigs
+rm -rf gateway-api
+
+# find the branch of gateway-api repo
+RELEASE_VERSION=${BUNDLE_VERSION/v/}
+RELEASE_VERSION=$(echo ${RELEASE_VERSION} | grep -oP "^\d+\.\d+")
+BRANCH="release-${RELEASE_VERSION}"
+echo "gateway-api repo branch: ${BRANCH}"
+
+# clone the branch that matched the installed CRDs version
+# branch release-1.0 -> bundle-version v1.0.0
+# branch release-1.1 -> bundle-version v1.1.1
+# branch release-1.2 -> bundle-version v1.2.1
+git clone --branch ${BRANCH} https://github.com/kubernetes-sigs/gateway-api
+cd gateway-api
+go mod vendor
+
+# modify default timeout of "MaxTimeToConsistency" to make tests pass on AWS
```

Comment:
Can you please elaborate what slows the test down on AWS specifically?

Reply:
Yes, the root cause is the DNS propagation of AWS ELB needs about extra 60s before the actual convergence process. The default MaxTimeToConsistency is 30s so we need to increase it to 90s here.
I ever proposed a change in PR https://github.com/kubernetes-sigs/gateway-api/pull/3542 and it was closed, seems many implementations use [kind](https://github.com/kubernetes-sigs/kind) cluster and metallb to run conformance and the LB can be available in short time, so they don't think the change is necessary.
But when we run the test on Cloud platforms (including Azure/GCP although they use IP instead of alias) it needs more time until the LB is reachable.

Reply:
Added one more line comment to explain this.